src/spartan/polys/masked_eq.rs

```diff
@@ -0,0 +1,150 @@
+//! `MaskedEqPolynomial`: Represents the `eq` polynomial over n variables, where the first 2^m entries are 0.
```

Comment:
Should this file be part of a predecessor PR that introduces boundsumcheck on W polynomial?

Reply:
I suggest we make a separate PR that we can merge quickly. This PR will include `masked_eq.rs` as well as the additional sum-check to check if W is padded appropriately. This will remove the need to invoke the evaluation argument twice. I think merging this before SuperNova makes the most sense.